### PR TITLE
CRM dashboard

### PR DESCRIPTION
**Pull request by saurabh-bhosale**
Second part of [Telephony services' Integration](https://github.com/frappe/erpnext/issues/13012).

1. Notifies an incoming call.
**Description:** Shows the notification of an incoming call to whoever the call is connected from the helpline/support team.
2. Updates the details on CRM Dashboard in realtime.

**Popup for an Incoming call:**
incoming_call_popup

- [ ] Gifs
- [ ] Docs

### CODE REVIEW

**Comment by rmehta** [comment]
@saurabh-bhosale any update. Please add relevant documentation too

https://github.com/frappe/erpnext/wiki/Contribution-Guidelines

File: erpnext/crm/doctype/crm_settings/crm_settings.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+			lead_doc = frappe.get_doc("Lead",contact_doc.get_link_for('Lead'))
+			dashboard_data = {
+				"title": "Lead",
+				"number": caller_no,
+				"route_link":lead_doc.name,
+				"name": lead_doc.lead_name or lead_doc.company_name,
+				"call_timestamp": frappe.utils.datetime.datetime.strftime(frappe.utils.datetime.datetime.today(), '%d/%m/%Y %H:%M:%S')
+			}
+
+		open_issues = frappe.get_all("Issue", filters = {"contact":contact_doc.get("name")}, fields=["*"])
```
Comment (rmehta) [inline erpnext/crm/doctype/crm_settings/crm_settings.py]: why select * ?

File: erpnext/erpnext_integrations/page/crm_dashboard/crm_dashboard.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+				console.log("comm",comm_details);
+				me.get_info(comm_details);
+			} else {
+				frappe.utils.notify(__("Incoming call"));
+			}
+		});
+	},
+	make_a_call: function(comm_details){
+		frappe.call({
+			method: "erpnext.erpnext_integrations.doctype.exotel_settings.exotel_settings.handle_outgoing_call",
```
Comment (rmehta) [inline erpnext/erpnext_integrations/page/crm_dashboard/crm_dashboard.js]: why is this hard coded?

It should be able to call any calling service

File: erpnext/crm/doctype/crm_settings/crm_settings.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+
+class CRMSettings(Document):
+	pass
+
+def make_popup(caller_no, comm_details):
```
Comment (rmehta) [inline erpnext/crm/doctype/crm_settings/crm_settings.py]: very large function

https://github.com/frappe/erpnext/wiki/Coding-Standards#break-functions--methods-more-than-10-lines-of-code

File: erpnext/crm/doctype/crm_settings/crm_settings.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	else:
+		popup_data = {
+			"title": "New Caller",
+			"number": caller_no,
+			"name": "Unknown",
+			"call_timestamp": frappe.utils.datetime.datetime.strftime(frappe.utils.datetime.datetime.today(), '%d/%m/%Y %H:%M:%S')
+		}
+		popup_data["route_link"] = str(comm_details.communication_name + "/" +
+			comm_details.get("communication_phone_no") + "/" +
+			comm_details.get("communication_exophone") +  "/" +
```
Comment (rmehta) [inline erpnext/crm/doctype/crm_settings/crm_settings.py]: exotel hardcoding

File: erpnext/crm/doctype/crm_settings/crm_settings.py (hunk 1 above)
Comment (rmehta) [reply erpnext/crm/doctype/crm_settings/crm_settings.py]: also add a limit, limit to last 5 issues

File: erpnext/public/js/telephony/telephony_console.html [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+			<li class="list-group-item">
+			<span class="h3"><i class="fa fa-user" style="margin-right: 20px"></i>
+				New Caller
+			</span>
+			</li>
+			<li class="list-group-item">
+				<span class="h3"><i class="fa fa-phone" style="margin-right: 20px"></i>{{ info.number }}</span>
+			</li>
+			<li class="list-group-item">
+				<button id="new_lead" class="btn btn-sm btn-primary">Create Lead</button>
```
Comment (rmehta) [inline erpnext/public/js/telephony/telephony_console.html]: public facing strings must be translatable

https://github.com/frappe/erpnext/wiki/Coding-Standards#user-facing-messages-must-be-translated

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by nabinhait** [comment]
@saurabh-bhosale Closing the PR for now. Please reopen after finishing the pending tasks.

**Pull request closed without merging**